packages/public/server/__tests-e2e__/login.ts

```diff
@@ -19,49 +19,129 @@
  * @link      https://github.com/serlo-org/serlo.org for the canonical source repository
  */
-import { pages, viewports, users, elements } from './_config'
+import { pages, viewports, elements, testingServerUrl } from './_config'
 import {
   clickForNewPage,
   getByPlaceholderText,
   getBySelector,
   getByText,
+  getByRole,
   goto,
 } from './_utils'
+import { ElementHandle } from 'puppeteer'
 
-const examplePages = ['/', '/math', '/math/geometry/triangles']
+describe('login process with correct data', () => {
+  test('correct password and correct user', async () => {
+    await page.setViewport(viewports.desktop)
+    const firstPage = await goto('/math')
+    // ?
+    expect(firstPage).toHaveUrlPath('/math')
 
-describe('login process', () => {
-  describe.each(users)('user is %p', (user) => {
-    test.each(examplePages)('start page is %p', async (startPath) => {
-      await page.setViewport(viewports.desktop)
-      const firstPage = await goto(startPath)
+    const loginPage = await elements
+      .getLoginButton(firstPage)
+      .then(clickForNewPage)
+    expect(loginPage).toHaveUrlPath('/auth/login')
+    const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
 
-      expect(firstPage).toHaveUrlPath(startPath)
+    await getByPlaceholderText(loginPage, inputUser).then((e) =>
+      e.type('login')
+    )
+    await getByPlaceholderText(loginPage, inputPassword).then((e) =>
+      e.type('123456')
+    )
+    const afterLoginPage = await getByText(loginPage, buttonLogin).then(
+      clickForNewPage
+    )
+    await expect(await elements.getLogoutButton(afterLoginPage)).toBeDefined()
 
-      const loginPage = await elements
-        .getLoginButton(firstPage)
-        .then(clickForNewPage)
+    const userPage = await elements
+      .getProfileButton(afterLoginPage)
+      .then(clickForNewPage)
 
-      expect(loginPage).toHaveUrlPath(pages.login.path)
+    expect(userPage).toHaveUrlPath('/user/me')
 
-      const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
-      await getByPlaceholderText(loginPage, inputUser).then((e) => e.type(user))
-      await getByPlaceholderText(loginPage, inputPassword).then((e) =>
-        e.type(pages.login.defaultPassword)
+    await expect(userPage).toMatchElement('h1', { text: 'login' })
+  })
+})
+
+describe('wrong login-data', () => {
+  const { buttonLogin, inputUser, inputPassword } = pages.login.identifier
+  let loginPage: ElementHandle
+
+  beforeEach(async () => {
+    await page.setViewport(viewports.desktop)
+    loginPage = await goto('/auth/login')
+  })
+
+  test('wrong password and correct user', async () => {
+    await getByPlaceholderText(loginPage, inputUser).then((e) =>
+      e.type('login')
+    )
+    await getByPlaceholderText(loginPage, inputPassword).then((e) =>
+      e.type('123')
+    )
+    const afterLoginPage = await getByText(loginPage, buttonLogin).then(
+      clickForNewPage
+    )
+    //Check if stayed on the page
```

Comment:
```suggestion
```
See https://github.com/labs42io/clean-code-typescript#prefer-self-explanatory-code-instead-of-comments

Reply:
You can make two test cases as in https://github.com/serlo/serlo.org-cloudflare-worker/blob/master/__tests__/frontend-proxy.ts#L175-L185 to describe your test code better. (Use an `beforeAll()` instead of `beforeEach()` because performance matters in e2e testing).